### DISCUSSION TITLE
What is the tdlib database encryption key used by nchat?

### DISCUSSION THREAD

**Discussion by i32333** [d99kris/nchat#163] [Q&A]
Hello folks, I am trying to write a program to search for public chats on Telegram. I want to use the TDLib encrypted database that nchat is currently using, so I need the DatabaseEncryptionKey. However, I have no experience in C++ development and couldn't find the DatabaseEncryptionKey in the nchat code. I would be grateful if someone could help me locate the DatabaseEncryptionKey in nchat's code.

**Comment by d99kris** [comment]
Hi @i32333 - I could be wrong, but I don't think the local tdlib database has a list of all public chat groups, as the number is more than 1.6 million according to https://tgstat.com/alerts (and the number is probably growing, so it would use a lot of bandwidth to keep all clients up-to-date with public groups, their descriptions, etc, if that was the case).

I would suggest to use some tdlib API to search for public chats instead.

**Reply by i32333** [reply]
I think I expressed it wrong. I want to use the tdlib API to implement a search for public chat functionality (and I did it) and then chat directly in nchat.

**Reply by i32333** [reply]
if I understand right, new chat should be searched before sending message. I would like to use the nchat session to search public chat and then send messages in nchat.

**Reply by d99kris** [reply]
I'm not very keen in facilitating use of tdlib database set up by nchat, mainly as it allows other applications use the nchat app id/key, and should someone use it to automate sending of messages (or other activity), Telegram may ban the nchat app id/key.

Of course the key (or how to read it out) is somewhere in the code and it's open source, but I don't think it benefits the nchat project from documenting it explicitly and publicly here. 

Having that said - if you could share the code you developed (finding public chats using the API) as a proof that you're using it for a good purpose, I would consider sharing how to figure out your private tdlib key (all users have different keys) privately, via email (`d99kris` `at` `gmail` `dot` `com`).

Btw, I assume you are creating this because nchat currently lacks built-in public chat search functionality?

**Comment by i32333** [comment]
Thank your for replying. Yes I actually wrote my program because nchat lacks built-in public chat searching. Though I'm using my program to start a new session for searching, and it's working fine for now,  I hope nchat would have the public chat searching soon.
